### ISSUE TITLE
Granting Espanso permission to access calendars on Mac

### ISSUE THREAD

**Issue by megclaypool** [espanso/espanso#650]
Hey there,

I'm trying to use Expanso to echo the output of icalBuddy.

Here's what I get when I run icalBuddy commands in the terminal:
```bash
$ icalBuddy -V
1.10.1

$ icalBuddy calendars
• Birthdays
  type: Birthday
  UID: ... and a long list of all my calendars which I'm not going to post here
```

However, when I use icalBuddy in a bash script which I'm calling using espanso, I'm not getting the same results.

My script contains:
```bash
echo "$(icalBuddy -V)"
echo "$(icalBuddy calendars)"
```
When I call it directly in the terminal using bash "path/to/my/script.sh", I get the same output as before. However, when I call it using espanso, I get:
```bash
1.10.1
          ← only a sad emptiness where my calendars should be 🙁
```

Looking at my espanso log, I see:
```
19:17:43 [ WARN] Script command reported error:
error: No calendars.
```

Another person seems to be having a [similar problem using icalBuddy with Alfred]( https://www.alfredforum.com/topic/12816-python-and-icalbuddy-calendar-workflow-stopped-working-in-alfred-4/), and the upshot was that they need to be able to grant Alfred permission to access the calendars, which is something the app must request from the user. The last post says that the issue was fixed by the Alfred dev: "Turns out it was Alfred not requesting permissions properly. "

Is it possible to grant espanso permission to access my calendars?

BTW, I'm running the latest macOS Big Sur (11.2.3).

**Comment by federico-terzi** [comment]
Hey @megclaypool,

Thanks for reporting this issue! I'm not very familiar with icalBuddy, but it seems it requires Automation privilege from the espanso side, which is not something the tool would normally request. Have you checked whether in the "Automation" tab of the Privacy settings (of your mac) espanso is listed?

Cheers :)

**Comment by megclaypool** [comment]
Nope, espanso doesn't appear there. Apparently the app has to open a dialog box requesting my permission to be added to that list -- I can't add it manually 🙁

**Comment by federico-terzi** [comment]
@megclaypool I see, thank you. Unfortunately, I don't have any idea on how to solve this problem, but if I discover something I'll let you know :)

**Comment by zacwest** [comment]
I believe you need to add the [calendar entitlement](https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_security_personal-information_calendars?language=objc) to the app for this. You're already using the Hardened Runtime because the app is notarized, which requires it.

For example, an entitlements file like:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>com.apple.security.personal-information.calendars</key>
	<true/>
</dict>
</plist>
```

Would be used as the parameter to the `codesign` command when signing for release, e.g.:

```bash
codesign --entitlements ./example.entitlements …
```

I wouldn't be surprised if you wanted to snag a few more -- addressbook, location perhaps. iTerm2's got a pretty [decent array](https://github.com/gnachman/iTerm2/blob/master/iTerm2.entitlements) for what it opts into using in the entitlements file which likely overlaps with Espanso usage.

**Comment by smeech** [comment]
Is this still a problem?

**Comment by megclaypool** [comment]
Honestly, I have no idea; because it wasn't working for me, I stopped using Espanso 🤷🏽‍♀️

**Comment by smeech** [comment]
I'm sorry to hear that, but hope you found a solution. It's possible the new release works better, and there are more of us involved in the project now, here and on discord.
I'll close this issue for now. If you want to try and work on it again, it can be reopened.

**Comment by zacwest** [comment]
Yes, this is still a problem. The code signing change I shared above will fix it. Unfortunately I cannot share a PR to do it as I cannot operate to test your CI infrastructure. Shouldn't be too bad though I would bet.
